flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/DynamicTableSourceSpecSerdeTest.java

```diff
@@ -238,6 +231,135 @@ public void testDynamicTableSourceSpecSerde() throws IOException {
                                                         put("p", "B");
                                                     }
                                                 }))));
-        return Arrays.asList(spec1, spec2);
+        return Stream.of(spec1, spec2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("testDynamicTableSinkSpecSerde")
+    public void testDynamicTableSourceSpecSerde(DynamicTableSourceSpec spec) throws IOException {
+        TableEnvironmentImpl tableEnv =
+                (TableEnvironmentImpl) TableEnvironment.create(inStreamingMode());
+
+        CatalogManager catalogManager = tableEnv.getCatalogManager();
```

Comment:
is it possible to unify with this sink test? I see a lot of duplicate code here.

Reply:
I tried to unify a bit some mocking, but there are some small differences between the two. Abstracting those will just make the test unreadable.